pkg/controller/resource_watcher.go

```diff
@@ -72,9 +75,54 @@ func (rw *ResourceWatcher) addPodController(controller PodController) {
 	rw.podControllers = append(rw.podControllers, controller)
 }
 
+func (rw *ResourceWatcher) clearAllControllers() {
+	// Set all controller slices to nil to clear them out
+	// https://github.com/golang/go/wiki/CodeReviewComments#declaring-empty-slices
+	rw.nodeControllers = nil
+	rw.nagControllers = nil
+	rw.podControllers = nil
+}
+
+func (rw *ResourceWatcher) StartElectedComponents(ctx context.Context) {
+	rw.log.Noticef("Starting elected components")
+
+	if rw.config.ParController.ShouldRun {
+		rw.addPodController(rw.parCtlr)
+	}
+
+	if rw.config.NagController.ShouldRun {
+		rw.addNodeController(rw.nagCtlr)
+		rw.addNagController(rw.nagCtlr)
+	}
+
+	if rw.config.PLController.ShouldRun {
+		rw.addNodeController(rw.plCtlr)
+		rw.addNagController(rw.plCtlr)
+		rw.addPodController(rw.plCtlr)
+	}
+
+	// Force a resync of all watched resources in case something was missed during leader switch
+	// All controllers are state-seeking so this is safe to do
+	if err := rw.nodeIndexer.Resync(); err != nil {
+		rw.log.Errorf("Error during resync after being elected")
+	}
+}
+
+func (rw *ResourceWatcher) StopElectedComponents() {
+	rw.log.Noticef("Stopping elected components")
+
+	// Remove all controllers to stop doing elected tasks
+	// Caches will continue to run in order to continue to provide data
+	// for webhook requests
+	rw.clearAllControllers()
+}
+
+func (rw *ResourceWatcher) ParController() *podassignment.Controller {
+	return rw.parCtlr
```

Comment:
If the par controller needs to be accessible outside of the resource watcher then it should really be injected into it rather than pulled out of it after it creates it.

Reply:
Trying to figure out a good way to do that.